rust/src/transformations/polars/make_cut/mod.rs

```diff
@@ -0,0 +1,141 @@
+use opendp_derive::bootstrap;
+use polars::prelude::*;
+
+use crate::core::{ExprFunction, Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{Context, ExprDomain, ExprMetric};
+use crate::error::*;
+use crate::metrics::{InsertDeleteDistance, SymmetricDistance};
+use crate::traits::{ExactIntCast, Number};
+
+#[bootstrap(ffi = false)]
+/// Make a Transformation that returns a `cut(breaks, labels, left_closed, include_breaks)` expression for a LazyFrame.
+///
+/// # Arguments
+/// * `input_domain` - Domain of the expression to be applied.
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared.
+/// * `breaks` - A list of unique cut points.
+/// * `labels` - Labels to assign to bins. If given, the length must be len(breaks) + 1.
+/// * `left_closed` - Whether intervals should be \[) instead of the default of (]
+/// * `include_breaks` - Include the the right endpoint of the bin each observation falls in. If True, the resulting column will be a Struct.
+///
+/// # Generics
+/// * `M` - Type of metric.
+///
+pub fn make_cut<MI>(
+    input_domain: ExprDomain<MI::Context>,
+    input_metric: MI,
+    breaks: Vec<f64>,
+    labels: Option<Vec<String>>,
+    left_closed: bool,
+    include_breaks: bool,
+) -> Fallible<Transformation<ExprDomain<MI::Context>, ExprDomain<MI::Context>, MI, MI>>
+where
+    MI: CutExprMetric + Sync + Send + 'static,
+    MI::Distance: Number + ExactIntCast<i64> + Clone + 'static,
+    (ExprDomain<MI::Context>, MI): MetricSpace,
+{
+    let output_domain = input_domain.clone();
+
+    let num_partitions = breaks.len();
+
+    Transformation::new(
+        input_domain.clone(),
+        output_domain,
+        Function::new_expr(move |expr| {
+            expr.clone().cut(
+                breaks.clone(),
+                labels.clone(),
+                left_closed.clone(),
+                include_breaks.clone(),
+            )
+        }),
+        input_metric.clone(),
+        input_metric.clone(),
+        StabilityMap::new_fallible(move |d_in: &MI::Distance| {
+            Ok(MI::Distance::exact_int_cast(
+                if <MI::Context as Context>::GROUPBY {
+                    input_metric.max_changed_partitions(num_partitions, &d_in)?
+                } else {
+                    1
+                },
+            )?)
+        }),
+    )
+}
+
+pub trait CutExprMetric: ExprMetric {
```

Comment:
I don't think you need this trait, you can just require the `M` to be an `ExprMetric`, and bound `M::InnerMetric` to be an `UnboundedMetric`. The associated type was useful when the type of the output metric was different, depending on what the input metric was. But that's not the case here.